src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -0,0 +1,595 @@
+				ArrayBuilder<string> properties = GetAttributeChilds (iterator.Current.SelectChildren ("property", string.Empty));
+
+				attributes.Add (new Attribute (attributeName, arguments, fields, properties));
+			}
+			return attributes;
+		}
+
+		ArrayBuilder<string> GetAttributeChilds (XPathNodeIterator iterator)
+		{
+			ArrayBuilder<string> childs = new ArrayBuilder<string> ();
+			while (iterator.MoveNext ()) {
+				childs.Add (iterator.Current.Value);
+			}
+			return childs;
+		}
+
+		void ProcessTypes (AssemblyDefinition assembly, XPathNodeIterator iterator)
+		{
+			while (iterator.MoveNext ()) {
+				XPathNavigator nav = iterator.Current;
+
+				if (!ShouldProcessSubstitutions (nav))
+					continue;
+
+				string fullname = GetFullName (nav);
+
+				if (IsTypePattern (fullname)) {
+					ProcessTypePattern (fullname, assembly, nav);
+					continue;
+				}
+
+				TypeDefinition type = assembly.MainModule.GetType (fullname);
+
+				if (type == null && assembly.MainModule.HasExportedTypes) {
+					foreach (var exported in assembly.MainModule.ExportedTypes) {
+						if (fullname == exported.FullName) {
+							var resolvedExternal = exported.Resolve ();
+							if (resolvedExternal != null) {
+								type = resolvedExternal;
+								break;
+							}
+						}
+					}
+				}
+
+				if (type == null) {
+					_context.LogMessage (MessageContainer.CreateWarningMessage ($"Could not resolve type '{fullname}' specified in {_xmlDocumentLocation}", 2008));
+					continue;
+				}
+
+				ProcessType (type, nav);
+			}
+		}
+
+		static bool IsTypePattern (string fullname)
+		{
+			return fullname.IndexOf ("*") != -1;
+		}
+
+		static Regex CreateRegexFromPattern (string pattern)
+		{
+			return new Regex (pattern.Replace (".", @"\.").Replace ("*", "(.*)"));
+		}
+
+		void MatchType (TypeDefinition type, Regex regex, XPathNavigator nav)
+		{
+			if (regex.Match (type.FullName).Success)
+				ProcessType (type, nav);
+
+			if (!type.HasNestedTypes)
+				return;
+
+			foreach (var nt in type.NestedTypes)
+				MatchType (nt, regex, nav);
+		}
+
+		void MatchExportedType (ExportedType exportedType, ModuleDefinition module, Regex regex, XPathNavigator nav)
+		{
+			if (regex.Match (exportedType.FullName).Success) {
+				TypeDefinition type = exportedType.Resolve ();
+				if (type != null) {
+					ProcessType (type, nav);
+				}
+			}
+		}
+
+		void ProcessTypePattern (string fullname, AssemblyDefinition assembly, XPathNavigator nav)
+		{
+			Regex regex = CreateRegexFromPattern (fullname);
+
+			foreach (TypeDefinition type in assembly.MainModule.Types) {
+				MatchType (type, regex, nav);
+			}
+
+			if (assembly.MainModule.HasExportedTypes) {
+				foreach (var exported in assembly.MainModule.ExportedTypes) {
+					MatchExportedType (exported, assembly.MainModule, regex, nav);
+				}
+			}
+		}
+
+		void ProcessType (TypeDefinition type, XPathNavigator nav)
+		{
+			if (!ShouldProcessSubstitutions (nav)) {
```

Comment:
Why do we have to check this again - as far as I can tell we already called it on the node.

Reply:
As described in the data-formats.md the feature switch can be applied to any element, so it may be the case in which the assembly doesn't have the feature switch but another element type/method/field/property does have it and it will only ignore those

Reply:
`ProcessTypes` call this on the type element already - this is repeating that call as far as I can tell.

Reply:
Got it! I fix it and added another one for nested types